extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1583,12 +1583,14 @@ export class Project implements ISqlProject {
 				const suppressMissingDependenciesErrorNode = currentNode.getElementsByTagName(constants.SuppressMissingDependenciesErrors);
 				const suppressMissingDependences = suppressMissingDependenciesErrorNode[0].childNodes[0].nodeValue === constants.True;
 
-				// remove this node
-				this.projFileXmlDoc!.documentElement.removeChild(currentNode);
+
 
 				// delete ItemGroup if there aren't any other children
-				if (this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.ArtifactReference).length === 0) {
+				if (this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.ArtifactReference).length === 1) {
```

Comment:
@kisantia I had to fix this logic since the new version of the lib now clears parentNode when a node is removed. 

But I had some questions about this since it seems like there's a couple potential bugs (that existed before too)

1. If the ItemGroup had other items in it then they'd also be deleted
2. If there are multiple item groups with ArtifactReferences then it won't delete the empty ItemGroups correctly

Does this behavior seem intended?

Reply:
These ArtifactReferences are something we don't expect people to be manually adding themselves in the sqlproj, so the current logic handles how SSDT adds them to the sqlproj, but it would be good to add extra checks to handle those cases just in case.

Reply:
Alright, I don't have the bandwidth to fix this now so I'll just add a comment here about the behavior and file an issue. That work for you?

Does the updated logic look fine? I figured I'd simplify it a bit since removing the parent node will remove the child node anyways.

Reply:
https://github.com/microsoft/azuredatastudio/issues/19641